lib/cisco_node_utils/vxlan_vtep_vni.rb

```diff
@@ -112,17 +112,36 @@ def ingress_replication
       config_get('vxlan_vtep_vni', 'ingress_replication', @get_args)
     end
 
+    def set_host_reachability(vtep_name, protocol)
+      # This is a helper method for the ingress_replication setter.
+      # In later versions of Nexus, a check was added to make sure
+      # the host_reachability setting is correct for the desired
+      # ingress_replication setting.
+      #
+      case protocol
+      when 'bgp'
+        host_reachability = 'evpn'
+      when 'static'
+        host_reachability = 'flood'
+      else
+        fail "Protocol #{protocol} currently not supported"
+      end
+      VxlanVtep.new(vtep_name).host_reachability = host_reachability
+    end
+
     def remove_add_ingress_replication(protocol)
       # Note: ingress-replication is not supported on all platforms.
       # Use to_s.empty check to also handle nil check.
       if ingress_replication.to_s.empty?
+        set_host_reachability(@set_args[:name], protocol)
```

Comment:
Maybe move this before if/else as it is exercise in both code paths

Reply:
Actually.. I tested it and it fails if I move it out of the if/else block.  The reason is that the two lines in the else block have to be executed first.

Reply:
would this work:

unless ingress_replication.to_s.empty?
  set_args_keys(state: 'no', protocol: ingress_replication)
  config_set('vxlan_vtep_vni', 'ingress_replication', @set_args)
end
set_host_reachability(@set_args[:name], protocol)
set_args_keys(state: '', protocol: protocol)
config_set('vxlan_vtep_vni', 'ingress_replication', @set_args)

Reply:
Yup.  That works.  Thanks for the suggestion.